avm/res/automation/automation-account/main.bicep

```diff
@@ -18,6 +18,9 @@ param skuName string = 'Basic'
 @description('Optional. The customer managed key definition.')
 param customerManagedKey customerManagedKeyType
 
+@description('Optional. List of credentials to be created in the automation account.')
+param credentials array = []
```

Comment:
We need a UDT for this to not expose the credential's secret

`arrays` don't support the `@secure()` flag - so we need to break it down into the User-Defined-Type and just like you configured it in the sub-module declare the actual secret as `@secure()`